test/adzerk/boot_reload/server_test.clj

```diff
@@ -4,16 +4,22 @@
 
 (deftest web-path-test
   (testing "Basic"
-    (is (= "/js/out/saapas/core.js"
-           (web-path "http:" "js/out/saapas/core.js" "target" nil))))
+    (is (= "js/out/saapas/core.js"
+           (web-path {:protocol "http:" :target-path "target"} "js/out/saapas/core.js"))))
 
-  (testing "Asset-path"
+  (testing "Output-dir"
     (is (= "js/out/saapas/core.js"
-           (web-path "http:" "public/js/out/saapas/core.js" nil "public")))
+           (web-path {:protocol "http:" :output-dir "public"} "public/js/out/saapas/core.js")))
 
     (is (= "js/out/saapas/core.js"
-           (web-path "http:" "public/js/out/saapas/core.js" nil "/public")))
+           (web-path {:protocol "http:" :output-dir "/public"} "public/js/out/saapas/core.js")))
 
     (is (= "public/js/out/saapas/core.js"
-           (web-path "http:" "public/js/out/saapas/core.js" nil "foobar")))
-    ))
+           (web-path {:protocol "http:" :output-dir "foobar"} "public/js/out/saapas/core.js"))))
+
+  (testing "Asset-path"
+    (is (= "js/saapas.out/saapas/core.js"
+           (web-path {:protocol "http:"
+                      :output-dir "resources/public/js/saapas.out"
+                      :asset-path "js/saapas.out"}
+                     "resources/public/js/saapas.out/saapas/core.js")))))
```

Comment:
Is this a real use case?

Writing files to resources directory doesn't make much sense, and even if it was the target directory it would make much sense.